testing/python/integration.py

```diff
@@ -410,22 +410,37 @@ def test_function_instance(pytester: Pytester) -> None:
     items = pytester.getitems(
         """
         def test_func(): pass
+
         class TestIt:
             def test_method(self): pass
+
             @classmethod
             def test_class(cls): pass
+
             @staticmethod
             def test_static(): pass
         """
     )
     assert len(items) == 4
+
     assert isinstance(items[0], Function)
     assert items[0].name == "test_func"
     assert items[0].instance is None
+
     assert isinstance(items[1], Function)
     assert items[1].name == "test_method"
     assert items[1].instance is not None
     assert items[1].instance.__class__.__name__ == "TestIt"
+
+    # Even class and static methods get an instance!
```

Comment:
I guess it is possible for this to break some plugin which is using `instance` to check if the method is a static/classmethod, but I suppose this is far fetched.

Reply:
You prompted me to check how plugins use `instance`. Overall it's almost not used, the only prominent case I found is pytest-asyncio, whose [usage](https://github.com/pytest-dev/pytest-asyncio/blob/b22d84e1f0d53920352be4c66d1b6c7f7a9ce005/pytest_asyncio/plugin.py#L296) is not problematic if I'm reading correctly and according to its tests.

pytest-asyncio however *was* broken by yesterday's removal of `FixtureDef.unittest` (#12089), for which I will send them a fix.